src/ExpireNFTv4.sol

```diff
@@ -0,0 +1,114 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.9;
+
+import "@openzeppelin/contracts/token/ERC721/ERC721.sol";
+import "@openzeppelin/contracts/token/ERC721/extensions/ERC721Enumerable.sol";
+import "@openzeppelin/contracts/token/ERC721/extensions/ERC721Burnable.sol";
+import "@openzeppelin/contracts/access/AccessControl.sol";
+import "@openzeppelin/contracts/utils/Counters.sol";
+import "@openzeppelin/contracts/utils/Base64.sol";
+
+contract ExpireNFTv4 is ERC721, ERC721Enumerable, ERC721Burnable, AccessControl {
```

Comment:
Is ERC721Enumerable required? I know it is pretty gas intensive, so i am curious as to why would you need that specifically.
From the docs: "IERC721Enumerable: Optional extension that allows enumerating the tokens on chain, often not included since it requires large gas overhead."

Reply:
Hi @7i7o Good advice! Anything to save some gas. I don't see why ERC721Enumerable is necessary. I will remove the import and contract declarations of it.

Reply:
Hi @7i7o I took ERCEnumerable out and redeployed, but now I cannot render the NFTs because there is no ERC721Supply extension that can be called. Do you see an easy, gas efficient way to get around this? Thank you again my friend.
![ERCEnumerable-error](https://user-images.githubusercontent.com/33005672/221250434-becbb857-fd3d-47e2-a582-da9a3d01f7d3.png)

Reply:
I think that Thirdweb Dashboard is looking for the `totalSupply()` function. That's why it's asking for `ERC721Supply` or `Enumerable` (which is overkill but includes the function).

Try implementing this function:
```
function totalSupply() public view returns (uint256) {
    return HERE_GOES_THE_TOTAL_SIZE_OF_THE_COLLECTION
}
```

And lmk if that solves it 😊

Reply:
Worked beautifully with 1/2 the gas to mint an NFT and that is considering that gas is currently on the expensive side. 0.0035 ETH compared to 0.00756 ETH with ERC721Enumerable,

Great upgrade! 
![image](https://user-images.githubusercontent.com/33005672/221256263-85e47c10-c42d-4be7-a8e9-5a7dc94f271d.png)